core-primitives/extrinsics-factory/src/lib.rs

```diff
@@ -84,20 +84,22 @@ where
 		let mut nonce_lock = self.nonce_cache.load_for_mutation()?;
 		let mut nonce_value = nonce_lock.0;
 
+		let extrinsic_params_builder = DefaultExtrinsicParamsBuilder::new()
+			.era(Era::Immortal, self.genesis_hash)
+			.tip(0);
+
```

Comment:
If they want to pay in a non-native asset, this line would need to be like this:

```
let extrinsic_params_builder = DefaultExtrinsicParamsBuilder::new()
			.era(Era::Immortal, self.genesis_hash)
			.tip(0)
			.of_asset(asset_id);
```

We could actually pass the builder from the outside into this function, and we might probably need to, if we want to have a configurable payment asset, specific to a set of extrinsics.

But this is out of scope for this PR, as we don't have an inherent need for this anyhow. Can you make an issue for this @echevrier?

Reply:
And add a todo here, that references said issue? :)

Reply:
This is the issue: https://github.com/integritee-network/worker/issues/794